### PR TITLE
Branch refactor code

### LINKED ISSUE #57: Code Review for @BenedictChannn
`src/main/java/cashleh/FileStorage.java`
* Line 47: Consider assigning the value of ".txt" to a variable then reference it as needed to reduce usage of magic strings
* Line 86: Likewise here, with the "]" delimiter, consider avoiding usage of magic literals
* Line 164, 166, 225, 227: Similarly, since you refer to the ")" delimiter a couple of times, consider assigning it to a variable to avoid magic literals
* Line 144, 205: The getIncome() and getExpense() methods do seem to be quite long and have some overlapping functionalities. Perhaps you can consider abstracting out some of the functionality to other methods to increase the readability of the code

`src/main/java/cashleh/Ui.java`
* Lines 60-65: since these values appear to be constants, perhaps you can consider defining them as such
* General: there are multiple instances of magic literals in this file, so do consider reducing dependency on magic literals, especially in the printHeader() function.

### PR DESCRIPTION
**Pull request by BenedictChannn**
Add sort by date
Fix magic literals
Tidy up code

### CODE REVIEW

File: src/main/java/cashleh/FileStorage.java [hunk 1]
```diff
@@ -142,127 +145,103 @@ public void writeToFile(IncomeStatement incomeStatement, ExpenseStatement expens
...
-        // Tokenize the input to get a map of income details
-        HashMap<String, String> incomeDetails = null;
         try {
-            incomeDetails = StringTokenizer.tokenize(input, incomePrefixes);
-        } catch (CashLehParsingException e) {
-            throw new CashLehFileCorruptedException(
-                "Wah, jialat! One or more necessary fields missing! Check your file again!");
+            if (categoryString == null && incomeDetails.length == 3) {
+                return new Income(description, amount);
+            } else if (incomeDetails.length == 3) {
```
Comment (mcmc101001) [inline src/main/java/cashleh/FileStorage.java]: i think this parts need better explanation, where does the number 3 come from?

File: src/main/java/cashleh/Ui.java [hunk 2]
```diff
@@ -84,6 +103,14 @@ public static void printStatement(String statementType, String[] texts) {
...
             totalExpense += (type.equals("Expense")) ? amount : 0;
 
+            // Format description text so that anything more than MAX_DESCRIPTION
+            // is replaced with "..."
+            String textReplacement = " ...";
+            if (description.length() > MAX_DESCRIPTION) {
+                description = description.substring(0, MAX_DESCRIPTION - textReplacement.length()
+                    - 2 * MIN_SPACE_PER_SIDE) + textReplacement;
+            }
+
```
Comment (mcmc101001) [inline src/main/java/cashleh/Ui.java:113]: good explaination!

**Review by mcmc101001 — COMMENTED** [review]
do update the developer guide and user guide!

File: src/main/java/cashleh/FileStorage.java (hunk 1 above)
Comment (BenedictChannn) [reply src/main/java/cashleh/FileStorage.java]: Thanks for pointing that out!

File: src/main/java/cashleh/Ui.java [hunk 3]
```diff
@@ -63,6 +68,20 @@ public static void printStatement(String statementType, String[] texts) {
         int descriptionSpace = 30;
         int amountSpace = 20;
         int categorySpace = 20;
+
```
Comment (danielpappa) [inline src/main/java/cashleh/Ui.java:71]: Great implementation

**Review by danielpappa — APPROVED** [review]
Very good code refactoring and improvement of levels of abstraction in the Ui class

**Review by Xuan127 — APPROVED** [review]
LGTM, would be good if there are more comments

**Pull request merged**